lib/auth/grpcserver.go

```diff
@@ -1944,14 +1960,34 @@ func (g *GRPCServer) GetMFADevices(ctx context.Context, req *proto.GetMFADevices
 		return nil, trace.Wrap(err)
 	}
 
-	devs, err := actx.authServer.GetMFADevices(ctx, actx.User.GetName())
+	// TODO(awly): mfa: remove secrets from MFA devices.
```

Comment:
Do you know Andrew meant here? Is it the key from TOTP devices?

If yes, then we should clear those out, specially since now we're adding new ways to list devices.

Reply:
> Do you know Andrew meant here? Is it the key from TOTP devices?

Double checked with andrew and he said yes and that u2f device data we store is not secret, regardless of this, the field that stores secret for TOTP is `Device` for both totp and u2f, so I set them both to nil if request is made without secrets. I used the same `withSecrets bool` approach as others like `GetUsers` and `GetSamlConnector`.